apps/bmd/src/utils/tallies.ts

```diff
@@ -6,10 +8,112 @@ import {
   VotesDict,
   YesNoVote,
 } from '@votingworks/types'
-import { CandidateVoteTally, Tally, YesNoVoteTally } from '../config/types'
+import {
+  CandidateVoteTally,
+  MsEitherNeitherTally,
+  Tally,
+  YesNoVoteTally,
+} from '../config/types'
 import { computeTallyForEitherNeitherContests } from './eitherNeither'
 import { getSingleYesNoVote } from './votes'
 
+const combineCandidateTallies = (
+  tally1: CandidateVoteTally,
+  tally2: CandidateVoteTally
+): CandidateVoteTally => {
+  const candidates: number[] = []
+  assert.strictEqual(tally1.candidates.length, tally2.candidates.length)
+  for (let i = 0; i < tally1.candidates.length; i++) {
+    candidates.push(tally1.candidates[i] + tally2.candidates[i])
+  }
+  const writeInVotes = [...tally1.writeIns]
```

Comment:
This copies the array, but not its contents. So the mutation you have inside the loop is still mutating the original.

Reply:
I'm actually refactoring how write ins work and remove this code in https://github.com/votingworks/vxsuite/pull/403

Reply:
just to be extra safe, I would suggest merging 403 into this first, so we can confirm this mutation no longer happens, before merging this to `main`. I'm being super careful here because we have been bitten before by mutating data structures, and so good to be extra extra careful.